### PR TITLE
Feature/ignore color option

### PR DESCRIPTION
**Pull request by lexhuismans**
This PR ignores the color option when provided as suggested in Issue #21.

It ignores all the options starting with `-c[olor=always|never|auto]` or `--c[olor=always|never|auto]`. Independent on where the option is provided.

Also added orderless option for `-4` and `-6` and `--version` for version, but this can be reverted.

### CODE REVIEW

**Review by ShadowJonathan — COMMENTED** [review]
Taking a look at the flag parsing, it feels a bit finnicky and manual to me, surely we can't use another flag-parsing library for this?

Also; I suggest emitting a warning in stderr regarding the current incompatibility of the color flag with iproute2mac, to make sure people aren't confused when there's no color, and that scripts relying on the stdout of the command dont get the garbage warning also piped to them.

File: src/ip.py [hunk 1]
```diff
@@ -727,17 +727,24 @@ def do_neigh_flush(argv, af):
...
-        argv.pop(0)
-    elif argv and argv[0] == "-4":
+        argv.remove("-6")
+    elif "-4" in argv:
         af = 4
-        argv.pop(0)
+        argv.remove("-4")
+    
+    # Ignore color option (-c --color)
+    for option in ["-c", "--c"]:
```
Comment (ShadowJonathan) [inline src/ip.py:738]:
```suggestion
    for option in ["-c", "--color"]:
```

`--color` doesnt match `--c`, it's exact string matching

Comment (lexhuismans) [reply src/ip.py:738]: Added the suggestion for stderr, --c and -c also match (-)-c[olor=always|never|auto] as we are checking if the arguments starts with --c or -c.

Hope to have some time to think about the flag parsing in the near future.

Comment (lexhuismans) [reply src/ip.py:738]: Not sure if I am missing something, but I already tested this. Below is an example. For you it didn't work?
Screenshot 2023-01-13 at 13 32 21

Comment (lexhuismans) [reply src/ip.py:738]: Not sure if I am missing something, but I tested this and it worked. For you it didn't work?

Comment (ShadowJonathan) [reply src/ip.py:738]: Ah no, but I see that only in the next few lines they're compared with startswith, which I missed

File: src/ip.py [hunk 2]
```diff
@@ -727,17 +727,24 @@ def do_neigh_flush(argv, af):
...
+        for arg in argv:
+            if arg.startswith(option):
+                perror("!Color option not available!")
+                argv.remove(arg)
 
     if not argv:
         return False
 
-    if argv[0] in ["-V", "-Version"]:
+    if argv[0] in ["-V", "-Version", "--version"]:
```
Comment (brona) [inline src/ip.py:747]: `--version` doesn't seem to be supported by iproute2, can you revert this change? Or what is your rationale?

```bash
user@hostname:~$ ip --version
Option "-version" is unknown, try "ip -help".
user@hostname:~$ ip --Version
ip utility, iproute2-ss190107
```

File: src/ip.py [hunk 3]
```diff
@@ -727,17 +727,24 @@ def do_neigh_flush(argv, af):
...
         af = 6
-        argv.pop(0)
-    elif argv and argv[0] == "-4":
+        argv.remove("-6")
+    elif "-4" in argv:
         af = 4
-        argv.pop(0)
+        argv.remove("-4")
+    
+    # Ignore color option (-c --color)
```
Comment (brona) [inline src/ip.py:737]: Please validate the flag according to iproute2 behavior, flag can be only added after `ip` and before e.g. `route`, see:
```bash
user@hostname:~# ip --color route
default via 172.22.0.1 dev ens2 onlink 
172.22.0.0/25 dev ens2 proto kernel scope link src 172.22.0.10 
user@hostname:~# ip route --color
Command "--color" is unknown, try "ip route help".
```

File: src/ip.py [hunk 4]
```diff
@@ -727,17 +727,24 @@ def do_neigh_flush(argv, af):
 def main(argv):
     # Detect Address family
     af = -1  # default / both
-    if argv and argv[0] == "-6":
+    if "-6" in argv:
```
Comment (brona) [inline src/ip.py:730]: Please see iproute2 behavior we are trying to match:

```bash
user@hostname:~# ip route -4
Command "-4" is unknown, try "ip route help".
```

File: src/ip.py [hunk 5]
```diff
@@ -727,17 +727,24 @@ def do_neigh_flush(argv, af):
...
         af = 4
-        argv.pop(0)
+        argv.remove("-4")
+    
+    # Ignore color option (-c --color)
+    for option in ["-c", "--c"]:
+        for arg in argv:
+            if arg.startswith(option):
+                perror("!Color option not available!")
+                argv.remove(arg)
```
Comment (brona) [inline src/ip.py:742]: Please use list comprehension or filter()

https://thispointer.com/python-remove-elements-from-a-list-while-iterating/

File: src/ip.py [hunk 6]
```diff
@@ -727,17 +727,24 @@ def do_neigh_flush(argv, af):
...
+    elif "-4" in argv:
         af = 4
-        argv.pop(0)
+        argv.remove("-4")
+    
+    # Ignore color option (-c --color)
+    for option in ["-c", "--c"]:
+        for arg in argv:
+            if arg.startswith(option):
+                perror("!Color option not available!")
```
Comment (brona) [inline src/ip.py:741]: This message is somewhat tricky, but I agree printing to stderr is likely going to be okay.

I suggest: `perror("iproute2mac: Color option is not implemented")`

**Review by brona — COMMENTED** [review]
Thanks a lot for your PR! I added few comments to be resolved.

File: src/ip.py (hunk 2 above)
Comment (lexhuismans) [reply src/ip.py:747]: Thank you for the extensive comments. The rationale for including `--version` (as well as not taking into account the order of the options) is to make it easier to use. 

There is an argument against this with it being confusing when shifting from Mac to Linux when some commands don't transfer one-on-one. If the goal is to match the behaviour exactly I will update the PR (and I could also add it to the contributing guidelines?)

Comment (brona) [reply src/ip.py:747]: With iproute2mac I am aiming for as much as identical behavior and flag compatibility with iproute on linux as possible. So here I would rather match the flag exactly. Thanks for understanding. 

And yes we should include this note in the contributing guidelines, after you PR I will extend them.

File: src/ip.py [hunk 7]
```diff
@@ -725,29 +725,34 @@ def do_neigh_flush(argv, af):
...
+    # Check all the options
+    while argv and argv[0].startswith("-"):
+        # Detect Address family
+        if argv[0] == "-6":
+            af = 6
+            argv.pop(0)
+        elif argv[0] == "-4":
+            af = 4
+            argv.pop(0)
+        elif "-color=always".startswith(argv[0]) \
```
Comment (brona) [inline src/ip.py:739]: Why not use just `argv[0].startswith("-color")` ?

**Review by brona — CHANGES_REQUESTED** [review]
Thanks, I think one last nit (see the comment).

And please could you rebase the PR into a single commit?  Thanks
